mimicpy/parsers/top.py

```diff
@@ -12,13 +12,16 @@ def fromDict(cls, df):
         keys = list(df.keys())
         df2 = df.copy()
         repeating = {}
-        for i in range(len(keys)):
+        i = 0
+        while i < len(keys):
```

Comment:
Why while and not for loop (or enumerate)?

Reply:
Basically I loop through the keys of df2. When a repeating element is found, it is deleted from df2. While looping, it will loop through the old key list and find a key that has been deleted. It will then throw an element not found error. So in every iteration, the keys list has to be updated to the new one.

This cannot be done in a for loop, and can only be done in a while loop.

Reply:
Honestly, I don't remember why I made a copy of df. I removed that, and the code still works.

Yes, the point is to skip the element that has been deleted.